### PR TITLE
feat: added alternative file transfer

### PR DESCRIPTION
**Pull request by Ninjagod1251**
We are going to accept a manifest format and return the same format with the compiled artifact.

This will help remix not have to convert and return files. It makes it simpler for both parties around

### CODE REVIEW

**Comment by sabotagebeats** [comment]
How can I test this for you @Ninjagod1251

**Review by sabotagebeats — CHANGES_REQUESTED** [review]
How can I test this for you?

File: main.py [hunk 1]
```diff
@@ -127,13 +160,27 @@ async def get_compiled_artifact(task_id: str) -> Dict:
...
+        path.parent.mkdir(parents=True, exist_ok=True)
+        path.write_text(source.fetch_content())
+
+    (project_root / ".build").mkdir()
+
+    (project_root / ".build" / "__local__.json").write_text(manifest.json())
+
     with config.using_project(project_root) as project:
         results[project_root.name] = project.extract_manifest()
     tasks[project_root.name] = TaskStatus.SUCCESS
```
Comment (fubuloubu) [inline main.py:186]:
```suggestion
         try:
             results[project_root.name] = project.extract_manifest()
             tasks[project_root.name] = TaskStatus.SUCCESS
         except Exception as e:
             results[project_root.name] = [str(e)]
             tasks[project_root.name] = TaskStatus.FAILED
```

**Comment by antazoey** [comment]
> How can I test this for you?

Testing instructions:

1. Launch app via `uvicorn main:app --reload`
2. Navigate to http://127.0.0.1:8000/docs in your browser
3. Click on the `compile` tab
4. It should pop open a little example and stuff

Previously without this PR, it would show a frozen loading circle and your browser would basically crash. LOL.
It is some bug in a Swagger Javascript U thing and has nothing to do with anything in Ape or even FastAPI. It is an OpenAPI (swagger) bug...

And thus our workaround here is to force the UI to load from an older version, at least that is what I think is happening... I kinda am just jumping in and helping, Chris if you have time, accurately explain what this workaround is doing.

> I add the workaround, can you test to see if you can reproduce the bug anymore? I can't reproduce it anymore

It works now! This is so cool, I am glad you got it working, thank you.

**Review by antazoey — APPROVED** [review]
nice work!

File: main.py [hunk 2]
```diff
@@ -127,13 +160,31 @@ async def get_compiled_artifact(task_id: str) -> Dict:
             status_code=404, detail="Task is not completed with Success status"
         )
     # TODO Debug why it is producing serialize_response raise ResponseValidationError( fastapi.exceptions.ResponseValidationError ) when you use return results[task_id] as a PackageManifest
-    return results[task_id].dict()
+    return results[task_id]
```
Comment (fubuloubu) [inline main.py:163]: These changes are important bug fixes

File: main.py [hunk 3]
```diff
@@ -127,13 +160,31 @@ async def get_compiled_artifact(task_id: str) -> Dict:
...
+
     with config.using_project(project_root) as project:
-        results[project_root.name] = project.extract_manifest()
-    tasks[project_root.name] = TaskStatus.SUCCESS
+         try:
+             results[project_root.name] = project.extract_manifest()
+             tasks[project_root.name] = TaskStatus.SUCCESS
+         except Exception as e:
+             results[project_root.name] = [str(e)]
+             tasks[project_root.name] = TaskStatus.FAILED
```
Comment (fubuloubu) [inline main.py:190]: These changes are important bug fixes

**Review by fubuloubu — APPROVED** [review]
It might be handy to pull out the bug fixes on another PR, I don't think the EF folks have had time to review this

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**